src/analysermodel.cpp

```diff
@@ -348,4 +349,39 @@ bool AnalyserModel::needAcothFunction() const
     return mPimpl->mNeedAcothFunction;
 }
 
+bool AnalyserModel::areEquivalentVariables(const VariablePtr &variable1,
+                                           const VariablePtr &variable2)
+{
+    // This is a cached version of the areEquivalentVariables() utility. Indeed,
+    // an AnalyserModel object refers to a static version of a model, which
+    // means that we can safely cache the result of a call to that utility. In
+    // turn, this means that we can speed up any feature (e.g., code generation)
+    // that also relies on that utility. When it comes to the key for the cache,
+    // we use the Cantor pairing function with the address of the two variables
+    // as parameters, thus ensuring the uniqueness of the key (see
+    // https://en.wikipedia.org/wiki/Pairing_function#Cantor_pairing_function).
+
+    auto v1 = reinterpret_cast<uint64_t>(variable1.get());
+    auto v2 = reinterpret_cast<uint64_t>(variable2.get());
```

Comment:
This won't work with emscripten. We cannot assume a fixed size of pointer. It is possible we leave this until Javascript compilation gets officially added though.

Reply:
I am happy to fix it now, but... why won't it work with emscripten and how should it be fixed?

Reply:
emscripten is only 32 bit so `uint64_t` won't work.

Reply:
As to the how I will have to remind myself.

Reply:
What about something like:
```c++
#if defined(__UINT64_TYPE__)
    auto v1 = reinterpret_cast<uint64_t>(variable1.get());
    auto v2 = reinterpret_cast<uint64_t>(variable2.get());
#elif defined(__UINT32_TYPE__)
    auto v1 = reinterpret_cast<uint32_t>(variable1.get());
    auto v2 = reinterpret_cast<uint32_t>(variable2.get());
#else
#    error "Unsupported architecture (only 32-bit and 64-bit architectures are supported)."
#endif
```

On 64-bit, both `__UINT32_TYPE__` and `__UINT64_TYPE__` are defined, but on 32-bit I would only expect `__UINT32_TYPE__` to be defined. So, it ought to work?

Reply:
I think I would prefer to go with `uintptr_t` do you see any reason not to? [integer types](https://en.cppreference.com/w/cpp/types/integer)

Reply:
No idea why I didn't use `uintptr_t` in the first place since, indeed, I can't see any reason not to.

Reply:
Done in commit e7a40a5.